docs/datatypes/tref.md

```diff
@@ -0,0 +1,223 @@
+---
+id: datatypes_tref
+title:  "TRef"
+---
+
+A `TRef[A]` is part of the `STM` bundle that represents a transactional reference that can perform reads and writes inside a transactional effect.
+
+`TRef` gives the basic machinery for others `SMT` transactional data types.
+
+`TRef[A]` where `A` is an immutable value.
+
+## Create a TRef
+
+Creating a `TRef` without committing its value:
+
+```scala mdoc:silent
+import zio.stm.{STM, TRef}
+
+val tVar: STM[Nothing, TRef[Int]] = TRef.make(10)
```

Comment:
`make` has been renamed to `apply`, hence it's `TRef(10)`.

Reply:
Arg, we should change this back (I am sorry I didn't notice that). In Scala, `TRef.apply` should give you a `TRef`, not a `TRef` inside `STM`. So it's a pervasive ZIO custom that `make` gives you something inside an effect.

Reply:
Gotcha, I'll polish these up asap.